src/interface_def.jl

```diff
@@ -16,30 +16,51 @@ end
 function field_allows_missing(field::Expr)
     field.head == :(::) && field.args[2] isa Expr &&
     field.args[2].head == :curly && field.args[2].args[1] == :Union &&
-    last(field.args[2].args) == :Missing
+    any(i->i == :Missing, field.args[2].args)
 end
 
-function field_type(field::Expr)
+function field_type(field::Expr, typename::String)
     if field.args[2] isa Expr && field.args[2].head == :curly && field.args[2].args[1] == :Union
-        return field.args[2].args[2]
+        if length(field.args[2].args) == 3 && (field.args[2].args[2] == :Missing || field.args[2].args[3] == :Missing)
+            return field.args[2].args[2] == :Missing ? field.args[2].args[3] : field.args[2].args[2]
+        else
+            # We return Any for now, which will lead to no type conversion
+            return Any
+        end
     else
         return field.args[2]
     end
 end
 
+function get_kwsignature_for_field(field::Expr)
+    fieldname = field.args[1]
+    fieldtype = field.args[2]
+    default_value = field_allows_missing(field) ? missing : :(error("You must provide a value for the $fieldname field."))
```

Comment:
`error("You must provide a value for the $fieldname field.")` should occur at macro expansion, right ?

Reply:
No, it should be part of the generated code. The idea is that we generate something like this to create a non-optional keyword argument:
```julia
function Foo(;bar=error("You must provide a value for bar"))
   ...
```